### PR TITLE
UI definition refactor

### PR DESCRIPTION
**Pull request by arkadius**
## Describe your changes

Changes in the definition API:
  * `/processDefinitionData/componentIds` endpoint is removed
  * `/processDefinitionData/*` response changes:
    * `services`, `sourceFactories`, `sinkFactories`, `customStreamTransformers` and `fragmentInputs` maps inside `processDefinition` were replaced by
      one `components` map with key in format `$componentType-$componentName` and moved into top level of response
    * `typesInformation` inside `processDefinition` was renamed into `classes`, moved into top level of response 
      and nested `clazzName` inside each element was extracted
    * `nodeId` field inside `edgesForNodes` was renamed into `componentId` in the flat `$componentType-$componentName` format
    * `defaultAsyncInterpretation` field was removed

Also, FE doesn't filter components by `categories` anymore - this job is already done on BE side. `categories` are still returned to FE in the response (it will be changed soon), but they aren't used by FE.

On the BE side:
* `ComponentIdProvider` moved back into designer module
* `ModelDefinitionWithComponentIds` is used locally in one place - in other places standard `ModelDefinition` is used instead
* `FragmentRepository` uses `FetchingProcessRepository` instead of its custom implementation
* `FragmentRepository` is filtering components by `processingType` instead of `category` 

## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

File: designer/client/src/common/ProcessUtils.ts [hunk 1]
```diff
@@ -171,36 +179,51 @@ class ProcessUtils {
...
+        const foundDefinition = components?.[this.findComponentId(node)];
         const emptyDefinition = {
             parameters: null,
             returnType: null,
         };
         return foundDefinition || emptyDefinition;
     };
 
-    private findDefinitionsForType = (processDefinition: ProcessDefinition, node?: NodeType): Record<string, NodeObjectTypeDefinition> => {
+    findComponentId = (node?: NodeType): string | null => {
```
Comment (wrzontek) [inline designer/client/src/common/ProcessUtils.ts:191]:
```suggestion
    resolveComponentId = (node?: NodeType): string | null => {
```
i'm not familiar with TS naming schemes, but to me `find` doesn't fit, it's all done on static information, no searching required

File: designer/client/src/common/ProcessUtils.ts [hunk 2]
```diff
@@ -244,15 +267,10 @@ class ProcessUtils {
...
-    findNodeDefinitionIdOrType = (node: NodeType) => this.findNodeDefinitionId(node) || node.type || null;
-
-    getNodeBaseTypeCamelCase = (node: NodeType) => node.type && node.type.charAt(0).toLowerCase() + node.type.slice(1);
-
     findNodeConfigName = (node: NodeType): string => {
         // First we try to find id of node (config for specific custom node by id).
-        // If it is falsy then we try to extract config name from node type (config for build-in components e.g. variable, join).
-        // If all above are falsy then it means that node is special process properties node without id and type.
-        return this.findNodeDefinitionId(node) || this.getNodeBaseTypeCamelCase(node) || "$properties";
+        // When is missing it means that node is special process properties node without id
```
Comment (wrzontek) [inline designer/client/src/common/ProcessUtils.ts:272]:
```suggestion
        // When it is missing it means that the node is a special process properties node without id
```
and the old comment above should be 
` // First we try to find the id of the node (configs for specific custom nodes are resolved by id).
`

articles are hard

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/process/fragment/FragmentRepository.scala [hunk 3]
```diff
@@ -1,132 +1,56 @@
...
+  final def fetchLatestFragmentsSync(processingType: ProcessingType)(implicit user: LoggedUser): List[FragmentDetails] =
+    Await.result(fetchLatestFragments(processingType), 10 seconds)
 
-  def loadFragments(versions: Map[String, VersionId], category: String): Set[FragmentDetails]
+  def fetchLatestFragments(processingType: ProcessingType)(implicit user: LoggedUser): Future[List[FragmentDetails]]
 
-  // FIXME: get by id in DB
-  def get(id: String): Option[FragmentDetails] = loadFragments().find(_.canonical.metaData.id == id)
+  // FIXME: async version should be used instead
+  final def fetchLatestFragmentSync(processName: ProcessName)(implicit user: LoggedUser): Option[FragmentDetails] =
```
[the remark this answers is not in the document]
Comment (arkadius) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/process/fragment/FragmentRepository.scala]: I changed to fragmentName, good point

File: designer/client/src/common/ProcessUtils.ts (hunk 1 above)
Comment (arkadius) [reply designer/client/src/common/ProcessUtils.ts]: Yeah, `find` is confusing in this context, but I think the `determine` word will be a little better than `resolve`, we use it quite often on the BE side in similiar context.

File: designer/client/src/common/ProcessUtils.ts (hunk 2 above)
Comment (arkadius) [reply designer/client/src/common/ProcessUtils.ts]: Yep, this comment had more problems than only articles, I rephrased it more deeply.

**Review by wrzontek — APPROVED** [review]
LGTM :+1:

**Pull request merged**